### ISSUE TITLE
Unable to use s3 cache exporter: failed to check file presence in cache

### ISSUE THREAD

**Issue by abatilo** [docker/buildx#1369]
Hi all! I'm trying to use the s3 remote cache and running into the following error:

```
ERROR: failed to solve: failed to check file presence in cache: operation error S3: HeadObject, https response error StatusCode: 400, RequestID: <>, HostID: <>, api error BadRequest: Bad Request
```

I'm running on an `Ubuntu 20.04.5 LTS` machine.

```terminal
⇒  docker --version
Docker version 20.10.20, build 9fdeb9c
```

I created my buildx agent with the following command:
```terminal
⇒  docker buildx create --name=builder --bootstrap --use --driver-opt=image=moby/buildkit:master
```

After creating the agent, I do an ECR login just to be safe also:
```
⇒  aws ecr get-login-password --region us-west-2 | docker login --username AWS --password-stdin <account>.dkr.ecr.us-west-2.amazonaws.com
```

I have my AWS config as follows:
```terminal
⇒  cat ~/.aws/config
[default]
region = us-west-2
output = json
```
```terminal
⇒  cat ~/.aws/credentials
[default]
aws_access_key_id = <>
aws_secret_access_key = <>
```
I'm using an AWS IAM user that has `AdministratorAccess` permissions.

And my `docker-bake.hcl` has a `target` that looks roughly like
```hcl
target "image_name" {
  context = "."
  tags = [
    "${ECR_REGISTRY}/iamge_name:latest",
  ]
  cache-from = ["type=s3,region=us-west-2,bucket=<bucket name>"]
  cache-to   = ["type=s3,region=us-west-2,bucket=<bucket name>,mode=max"]
}
```
The bucket exists, I'm able to `aws s3 cp` files into it. I'm kind of out of ideas for what to try next. Any ideas?

I'd love to provider any other information.

**Comment by abatilo** [comment]
Also opened an issue at https://github.com/moby/buildkit/issues/3211

**Comment by MarcoGlauser** [comment]
We had the same issue. After some experimenting, we found that the 400 error code is just a generic code used by aws and that the underlying issue was an authentication issue.
And the root cause was mentioned in the docs:
> AWS Authentication:
>* Any system using environment variables / config files supported by the [AWS Go SDK](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html). **The configuration must be available for the buildkit daemon, not for the client.**

Adding the access_key_id and secret_access_key to the command resolved the issue. However this doesn't feel very clean. Is there another way to provide the credentials to the buildkit daemon?

**Comment by crazy-max** [comment]
Client-side support is already implemented https://github.com/docker/buildx/pull/1294 but not yet released. You need to [build and install buildx](https://github.com/docker/buildx#building) from master branch and it should work.

**Comment by MarcoGlauser** [comment]
https://github.com/docker/buildx/pull/1294 has been included in the 0.10.0 release of buildx :tada:
